lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -1060,6 +1060,23 @@ bool WitnessChecker::findBestWitness(
   }
 
   if (numViable == 0) {
+    // Assume any missing value witnesses for a conformance in a textual
+    // interface can be treated as opaque.
+    // FIXME: ...but we should do something better about types.
```

Comment:
If we don’t expect to generate textual interfaces for Swift 4 modules, we can make it an error in Swift 5 mode.

Reply:
We may still want to to help with incremental builds, but yes.